packages/core/src/transport/sendWithRetryStrategy.spec.ts

```diff
@@ -190,50 +190,68 @@ describe('sendWithRetryStrategy', () => {
+          sendStub.respondWith(0, { status, type })
+          expect(state.queuedPayloads.size()).toBe(1)
+
+          sendRequest()
+          expect(state.queuedPayloads.size()).toBe(2)
+          sendRequest()
+          expect(state.queuedPayloads.size()).toBe(3)
+        })
 
-        sendRequest()
-        expect(state.queuedPayloads.size()).toBe(2)
-        sendRequest()
-        expect(state.queuedPayloads.size()).toBe(3)
-      })
+        it('should send queued requests if another ongoing request succeed', () => {
+          sendRequest()
+          sendRequest()
+          sendStub.respondWith(0, { status, type })
+          expect(state.bandwidthMonitor.ongoingRequestCount).toBe(1)
+          expect(state.queuedPayloads.size()).toBe(1)
 
-      it('should send queued requests if another ongoing request succeed', () => {
-        sendRequest()
-        sendRequest()
-        sendStub.respondWith(0, { status })
-        expect(state.bandwidthMonitor.ongoingRequestCount).toBe(1)
-        expect(state.queuedPayloads.size()).toBe(1)
+          sendRequest()
+          expect(state.bandwidthMonitor.ongoingRequestCount).toBe(1)
+          expect(state.queuedPayloads.size()).toBe(2)
 
-        sendRequest()
-        expect(state.bandwidthMonitor.ongoingRequestCount).toBe(1)
-        expect(state.queuedPayloads.size()).toBe(2)
-
-        sendStub.respondWith(1, { status: 200 })
-        expect(state.bandwidthMonitor.ongoingRequestCount).toBe(2)
-        expect(state.queuedPayloads.size()).toBe(0)
-      })
+          sendStub.respondWith(1, { status: 200 })
+          expect(state.bandwidthMonitor.ongoingRequestCount).toBe(2)
+          expect(state.queuedPayloads.size()).toBe(0)
+        })
 
-      it('should add retry info to payloads', () => {
-        sendRequest()
+        it('should add retry info to payloads', () => {
+          sendRequest()
 
-        sendStub.respondWith(0, { status })
-        expect(state.queuedPayloads.first().retry).toEqual({ count: 1, lastFailureStatus: status })
+          sendStub.respondWith(0, { status, type })
+          expect(state.queuedPayloads.first().retry).toEqual({
+            count: 1,
+            lastFailureStatus: status,
+            lastFailureType: type,
+          })
 
-        clock.tick(INITIAL_BACKOFF_TIME)
+          clock.tick(INITIAL_BACKOFF_TIME)
 
-        sendStub.respondWith(1, { status })
-        expect(state.queuedPayloads.first().retry).toEqual({ count: 2, lastFailureStatus: status })
-      })
+          sendStub.respondWith(1, { status, type })
+          expect(state.queuedPayloads.first().retry).toEqual({
+            count: 2,
+            lastFailureStatus: status,
+            lastFailureType: type,
+          })
+        })
+      } else {
+        it('should not queue the payload for retry', () => {
+          sendRequest()
+          sendStub.respondWith(0, { status, type })
+          expect(state.queuedPayloads.size()).toBe(0)
+        })
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Since there is only one case where `expectRetry` is false, and test cases are completely different (no common code executed), what about declaring a separate test case for it?

Reply:
I was thinking about the next PR that should introduce other cases around `onLine` status to decide if we should retry or not.
So having all the cases described in the same place and running the different expectations depending on if we shold retry or not seemed convenient.
wdyt?

Reply:
Alright, in this case it seems fine!